fast_rnnt/python/fast_rnnt/rnnt_loss.py

```diff
@@ -657,28 +679,41 @@ def get_rnnt_prune_ranges(
     """
     (B, S, T1) = px_grad.shape
     T = py_grad.shape[-1]
+
+    is_regular = T1 != T
+
     assert T1 in [T, T + 1], T1
     S1 = S + 1
     assert py_grad.shape == (B, S + 1, T), py_grad.shape
     assert boundary.shape == (B, 4), boundary.shape
 
-    assert S >= 1, S
-    assert T >= S, (T, S)
+    validate_st_lengths(S, T, is_regular, boundary)
+
+    # adjust s_range if S >> T in regular case
+    if is_regular:
+        Ss = boundary[:, 2]
+        Ts = boundary[:, 3]
+        s_range_min = (Ss - 2).div(Ts, rounding_mode="trunc").add(3).max().item()
```

Comment:
Could you explain this formula, I don't get you idea, sorry!

I think it should be `(s_range - 1) * Ts + 1 >= Ss + 1` and the simplified version is `s_range_min = Ss // Ts + 2`.  BTW, I checked several cases of both formula, the results are the same.

Reply:
Thanks for checking that, your formula is simpler while still working correctly.

Reply:
No, I think my formula won't work if `Ts== 1`,  you can double check that. Thanks!

Reply:
Wait, It seems that your formula has a minimum `s_range_min` 3, which means we can't train a model with `s_range==2`. I think we can use `Ss // Ts + 2` and add a constraint `if s_range_min > S + 1: s_range_min = S + 1`.

Reply:
Thanks, very nice catch. The correct formula is 
`(Ss - 1) // Ts + 2` and it works for all cases. I updated the code accordingly.